### PR TITLE
feat(cms): icky 735 refactored cms contentstack components

### PR DESCRIPTION
**Pull request by TriptiVermaXP**

### CODE REVIEW

**Review by kibo-geetanshuchhabra — APPROVED** [review]

**Review by kibo-sushant — APPROVED** [review]

File: components/CmsComponent.vue [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        mapDataToProps: (data) => {
+          return {
+            homePageProduct: cmsGetters.getHomePageProducts(data),
+          }
+        },
+      },
+      hero_carousel: {
+        component: HeroCarousel,
+        mapDataToProps: (data) => {
+          return {
```
Comment (kibo-susantasamanta) [inline components/CmsComponent.vue]: Return can be done without return key here.

File: components/CmsComponent.vue [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            carouselItem: cmsGetters.getHeroCarousel(data),
+          }
+        },
+      },
+      large_promo_blocks: {
+        component: PromoBlocks,
+        mapDataToProps: (data) => {
+          return {
```
Comment (kibo-susantasamanta) [inline components/CmsComponent.vue]: Return can be done without return key here.

File: components/CmsComponent.vue [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            promoBlocks: cmsGetters.getPromoBlocks(data),
+          }
+        },
+      },
+      small_promo_blocks: {
+        component: PromoBlocks,
+        mapDataToProps: (data) => {
+          return {
```
Comment (kibo-susantasamanta) [inline components/CmsComponent.vue]: Return can be done without return key here.

File: components/CmsComponent.vue [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        mapDataToProps: (data) => cmsGetters.getProductRecommendations(data),
+      },
+      customers_also_bought: {
+        component: KiboProductCarousel,
+        mapDataToProps: (data) => cmsGetters.getCustomersAlsoBought(data),
+      },
+    }
+
+    const name = Object.keys(content)[0]
+    const ComponentMapping = props.ComponentMap || DefaultComponentMap
```
Comment (kibo-susantasamanta) [inline components/CmsComponent.vue]: First letter of the variable is capital, is this mandatory here? Same for DefaultComponentMap.

File: components/HeroCarousel.vue [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  },
+}
+</script>
+
+<style lang="scss" scoped>
+@import "~@storefront-ui/shared/styles/components/organisms/SfHero.scss";
+
+::v-deep .sf-hero {
+  &__slides {
+    height: calc(85vh - 3.438rem);
```
Comment (kibo-susantasamanta) [inline components/HeroCarousel.vue:64]: Instead rem can we not use var?

File: components/HeroCarousel.vue [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+    @include for-desktop {
+      align-items: center;
+    }
+
+    &__wrapper {
+      display: flex;
+      align-items: center;
+      background-color: var(--c-white);
+      margin: 2rem 1.5rem;
```
Comment (kibo-susantasamanta) [inline components/HeroCarousel.vue]: Instead rem can we not use var?

File: components/HeroCarousel.vue [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      height: auto;
+      min-height: auto;
+      padding: var(--spacer-lg);
+      text-align: center;
+    }
+
+    &__subtitle {
+      --hero-item-subtitle-font-size: 2rem;
```
Comment (kibo-susantasamanta) [inline components/HeroCarousel.vue]: I think we can use var to get front-weight, size etc all places where ever we can.

File: components/PromoBlocks.vue [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+.large-content {
+  &__header {
+    color: var(--_c-green-primary);
+    font-size: var(--spacer-base);
+    line-height: calc(var(--spacer-2xs) * 7.125);
+    text-align: left;
+    font-weight: bold;
+    min-width: 289px;
```
Comment (kibo-susantasamanta) [inline components/PromoBlocks.vue]: Can we not use var here?

File: components/PromoBlocks.vue [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+  &__tiles {
+    margin-bottom: calc(var(--spacer-2xs) * 9);
+
+    @include for-desktop {
+      width: 48%;
+    }
+
+    ::v-deep .sf-image {
+      height: 400px;
```
Comment (kibo-susantasamanta) [inline components/PromoBlocks.vue]: should use var ?

File: components/PromoBlocks.vue [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+  &__tiles {
+    margin-bottom: calc(var(--spacer-2xs) * 9);
+    width: 45%;
+
+    @include for-desktop {
+      width: 24%;
+    }
+
+    ::v-deep .sf-image {
+      height: 260px;
```
Comment (kibo-susantasamanta) [inline components/PromoBlocks.vue]: should use var?

File: components/SmallBanner.vue [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+<style lang="scss" scoped>
+@import "~@storefront-ui/shared/styles/components/organisms/SfHero.scss";
+
+.small-banner {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  background-color: #a12e87;
```
Comment (kibo-susantasamanta) [inline components/SmallBanner.vue:44]: Should use color from variables.

**Review by kibo-susantasamanta — CHANGES_REQUESTED** [review]
Components can be  typescript  supported by using lang='ts' inside script tag.

**Review by kibo-susantasamanta — APPROVED** [review]

File: components/CmsComponent.vue (hunk 1 above)
Comment (TriptiVermaXP) [reply components/CmsComponent.vue]: done

File: components/CmsComponent.vue (hunk 2 above)
Comment (TriptiVermaXP) [reply components/CmsComponent.vue]: done

File: components/CmsComponent.vue (hunk 3 above)
Comment (TriptiVermaXP) [reply components/CmsComponent.vue]: done

File: components/HeroCarousel.vue (hunk 5 above)
Comment (TriptiVermaXP) [reply components/HeroCarousel.vue:64]: As discussed not changed

File: components/HeroCarousel.vue (hunk 6 above)
Comment (TriptiVermaXP) [reply components/HeroCarousel.vue]: done

File: components/HeroCarousel.vue (hunk 7 above)
Comment (TriptiVermaXP) [reply components/HeroCarousel.vue]: done

File: components/SmallBanner.vue (hunk 11 above)
Comment (TriptiVermaXP) [reply components/SmallBanner.vue:44]: As discussed, not changed

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**